myapp/app/controllers/users_controller.rb

```diff
@@ -31,14 +33,25 @@ def create_user
   end
 
   def edit 
+    @user = find_user_by_id
+    redirect_to admin_path, notice: 'User not found!' unless @user
+    @myself = params[:id].to_i == session[:user_id]
+  end
+
+  def update_info
```

Comment:
[MUST]
Could you add validation logic in server-side for controlling number of admin users?

> 管理ユーザが1人もいなくならないように削除の制御をしましょう

Reply:
Checked server-side validation!
[238c1b2](https://github.com/Fablic/training/pull/1677/commits/238c1b20c2e6961399b199b46588d4d6d6b61403)